src/main/java/com/teragrep/pth_06/planner/bloomfilter/TokenizedValue.java

```diff
@@ -43,39 +43,55 @@
  * Teragrep, the applicable Commercial License may apply to this file if you as
  * a licensee so wish it.
  */
-package com.teragrep.pth_06.planner;
+package com.teragrep.pth_06.planner.bloomfilter;
 
 import com.teragrep.blf_01.Token;
 import com.teragrep.blf_01.Tokenizer;
 
 import java.io.ByteArrayInputStream;
 import java.nio.charset.StandardCharsets;
 import java.util.HashSet;
+import java.util.Objects;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 public final class TokenizedValue {
 
-    public final String value;
+    private final String value;
+    private final Set<Token> tokenSet;
 
     public TokenizedValue(String value) {
+        this(
+                value,
+                new HashSet<>(new Tokenizer(32).tokenize(new ByteArrayInputStream(value.getBytes(StandardCharsets.UTF_8))))
```

Comment:
not 100% sure if calling tokenizer in constructor is optimal, perhaps it should only be done when tokens are required?

Reply:
Refactored that tokenizer is called only when tokens are needed.